R/course.R

```diff
@@ -0,0 +1,324 @@
+#' use_course("https://github.com/r-lib/rematch2/archive/master.zip")
+#' use_course("http://github.com/r-lib/rematch2/zipball/master/")
+#' }
+use_course <- function(url, destdir = NULL) {
+  zipfile <- download_zip(url, destdir = destdir, pedantic = TRUE)
+  tidy_unzip(zipfile)
+}
+
+#' Download a ZIP file
+#'
+#' Special-purpose function to download a ZIP file and automatically determine
+#' the file name, which often determines the folder name after unpacking.
+#' Developed for use in live teaching, with DropBox and GitHub as primary
+#' targets, possibly via shortlinks. Both platforms offer a way to download an
+#' entire folder or repo as a ZIP file, with information about the original
+#' folder or repo transmitted in the `Content-Disposition` header. In the
+#' absence of this header, a filename is generated from the input URL. In either
+#' case, the filename is sanitized.
+#'
+#' @section DropBox:
+#'
+#' To make a folder available for ZIP download, create a shared link for it:
+#' * <https://www.dropbox.com/help/files-folders/view-only-access>
+#'
+#' A shared link will have this form:
+#' ```
+#' https://www.dropbox.com/sh/12345abcde/6789wxyz?dl=0
+#' ```
+#' Replace the `dl=0` at the end with `dl=1` to create a download link. The ZIP
+#' download link will have this form:
+#' ```
+#' https://www.dropbox.com/sh/12345abcde/6789wxyz?dl=1
+#' ```
+#' This download link (or a shortlink that points to it) is suitable as input
+#' for `download_zip()`. After one or more redirections, this link will
+#' eventually lead to a download URL. For more details, see
+#' <https://www.dropbox.com/help/desktop-web/force-download> and
+#' <https://www.dropbox.com/en/help/desktop-web/download-entire-folders>.
+#'
+#' @section GitHub:
+#'
+#' Click on the repo's "Clone or download" button, to reveal a "Download ZIP"
+#' button. Capture this URL, which will have this form:
+#' ```
+#' https://github.com/r-lib/usethis/archive/master.zip
+#' ```
+#' This download link (or a shortlink that points to it) is suitable as input
+#' for `download_zip()`. After one or more redirections, this link will
+#' eventually lead to a download URL. Here's an alternative link that also leads
+#' to ZIP download, albeit with a different filenaming scheme:
+#' ```
+#' http://github.com/r-lib/usethis/zipball/master/
+#' ```
+#'
+#' @param url Download link for the ZIP file, possibly behind a shortlink or
+#'   other redirect. See Details.
+#' @param destdir Path to existing local directory where the ZIP file will be
+#'   stored. Defaults to current working directory.
+#' @param pedantic Logical. When `TRUE` (default) and `destdir = NULL` and in an
+#'   interactive session, the user is told where the ZIP file will be stored. If
+#'   happy, user can elect to proceed. Otherwise, user can abort and try again
+#'   with the desired `destdir`. Intentional friction.
+#'
+#' @return Path to downloaded ZIP file
+#' @keywords internal
+#' @family download functions
+#' @export
```

Comment:
I guess should be exported?

Reply:
I think you can leave internal for this release, if you want

Reply:
`download_zip() and `tidy_unzip()` are now unexpected. Moved their docs to a common dummy topic. 99148615d9f7d0d44eba3779a0d58be9cb428382